src/main/java/Duke/System/TaskList.java

```diff
@@ -0,0 +1,116 @@
+package Duke.System;
+
+import Duke.DukeException.DukeException;
+import Duke.Tasks.Task;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public void add(Task task) {
+        this.tasks.add(task);
+    }
+
+    public void delete(int i) throws DukeException {
+        if (!this.get(i).isEmptyTask()) {
+            this.tasks.remove(i - 1);
+        } else {
+            throw new DukeException("No such task found!");
+        }
+    }
+
+    public Task get(int i) throws DukeException {
+        if (tasks.isEmpty()) {
+            throw new DukeException("TaskList is empty!");
+        } else if (i < 1 || i > tasks.size()) {
+            throw new DukeException("Requested task does not exist!");
+        } else {
+            return tasks.get(i - 1);
+        }
+    }
+
+    public int getNumOfTasks() {
+        return tasks.size();
+    }
+
+    public String tasksAsString() throws DukeException {
```

Comment:
You could refactor this function to the Ui file instead.  Otherwise really liked your implementation of the TaskList file.